plugins/modules/deepsec_firewall_rules.py

```diff
@@ -0,0 +1,674 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: deepsec_firewall_rules
+short_description: Manages Firewall Rule resource module
+description: Firewall rule details.
+version_added: 2.0.0
+options:
+  config:
+    description: A dictionary of Firewall Rules options
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description: Name of the firewall rule. Searchable as String.
+        type: str
+      description:
+        description: Description of the firewall rule. Searchable as String.
+        type: str
+      action:
+        description: Action of the packet filter. Searchable as Choice.
+        type: str
+        choices:
+        - log-only
+        - allow
+        - deny
+        - force-allow
+        - bypass
+      priority:
+        description: Priority of the packet filter. Searchable as Choice.
+        type: str
+        choices:
+        - '0'
+        - '1'
+        - '2'
+        - '3'
+        - '4'
+      direction:
+        description: Packet direction. Searchable as Choice.
+        type: str
+        choices:
+        - incoming
+        - outgoing
+      frame_type:
+        description: Supported frame types. Searchable as Choice.
+        type: str
+        choices:
+        - any
+        - ip
+        - arp
+        - revarp
+        - ipv4
+        - ipv6
+        - other
+      frame_number:
+        description: Ethernet frame number. Only required for FrameType "other". Searchable
+          as Numeric.
+        type: int
+      frame_not:
+        description: Controls if the frame setting should be inverted. Set to true
+          to invert. Searchable as Boolean.
+        type: bool
+      protocol:
+        description: Protocol. Searchable as Choice.
+        type: str
+        choices:
+        - any
+        - icmp
+        - igmp
+        - ggp
+        - tcp
+        - pup
+        - udp
+        - idp
+        - nd
+        - raw
+        - tcp-udp
+        - icmpv6
+        - other
+      protocol_number:
+        description: Two-byte protocol number. Searchable as Numeric.
+        type: int
+      protocol_not:
+        description: Controls if the protocol setting should be inverted. Set to true
+          to invert. Searchable as Boolean.
+        type: bool
+      source_iptype:
+        description: Source IP type. Default is "any". Searchable as Choice.
+        type: str
+        choices:
+        - any
+        - masked-ip
+        - range
+        - ip-list
+        - single
+        - multiple
+      source_ipvalue:
```

Comment:
Can we move all the `source_*` and `destination_*` keys under a dict with the respective keys as parent?

```
source:
  ipvalue:
  ipmask:
destination:
   ipvalue:
   ipmask:
```

Reply:
Here for security platforms, we generate module from their REST API and thus replicate the module params from REST doc, coz of which we're not planning on making structural changes to available params.